### PR TITLE
Test-runner file for hook configuration

### PR DESCRIPTION
**Pull request by shilman**
## What I did

- [x] Added a new test-runner config file &  types
- [x] Updated the documentation

Tried to remove the old API but couldn't get it working so I went back to the old way.

<details>
  <summary>📦 Published PR as canary version: <code>0.0.3-canary.53.755f06f.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/test-runner@0.0.3-canary.53.755f06f.0
  # or 
  yarn add @storybook/test-runner@0.0.3-canary.53.755f06f.0
  ```
</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -255,10 +256,40 @@ The test runner renders a story and executes its [play function](https://storybo
...
+
+module.exports = {
+  setup() {
+    expect.extend({ toMatchImageSnapshot });
+  },
+  async postRender(page, context) {
+    const image = await page.screenshot();
+    expect(image).toMatchImageSnapshot({
+      customSnapshotsDir,
+      customSnapshotIdentifier: context.id,
```
Comment (yannbf) [inline README.md:285]: Do you think we should add the failureThreshold here in the recipe?

File: README.md [hunk 2]
```diff
@@ -255,10 +256,40 @@ The test runner renders a story and executes its [play function](https://storybo
...
+    });
+  },
+};
+```
+
```
Comment (yannbf) [inline README.md:290]: Do we want to mention at some point that if the user prefers typescript they can also use, and if so there is a type for that config in @storybook/test-tunner?

File: README.md [hunk 3]
```diff
@@ -17,9 +17,10 @@ Storybook test runner turns all of your stories into executable tests.
     - [2. Running against locally built Storybooks in CI](#2-running-against-locally-built-storybooks-in-ci)
   - [Experimental test hook API](#experimental-test-hook-api)
     - [Image snapshot recipe](#image-snapshot-recipe)
+    - [Render lifecycle](#render-lifecycle)
   - [Troubleshooting](#troubleshooting)
-    - [The test runner seems flaky and keeps timing out](#the-test-runner-seems-flaky-and-keeps-timing-out)
-    - [Adding the test runner to other CI environments](#adding-the-test-runner-to-other-ci-environments)
+      - [The test runner seems flaky and keeps timing out](#the-test-runner-seems-flaky-and-keeps-timing-out)
+      - [Adding the test runner to other CI environments](#adding-the-test-runner-to-other-ci-environments)
```
Comment (yannbf) [inline README.md:23]: Was this change intended?
```suggestion
    - [The test runner seems flaky and keeps timing out](#the-test-runner-seems-flaky-and-keeps-timing-out)
    - [Adding the test runner to other CI environments](#adding-the-test-runner-to-other-ci-environments)
```

File: playwright/jest-setup.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const { getTestRunnerConfig, setPreRender, setPostRender } = require('../dist/cjs');
+
+const testRunnerConfig = getTestRunnerConfig(process.env.STORYBOOK_CONFIG_DIR);
+if (testRunnerConfig) {
+  if (testRunnerConfig.setup) {
+    testRunnerConfig.setup();
+  }
+  if (testRunnerConfig.preRender) {
+    setPreRender(testRunnerConfig.preRender);
```
Comment (yannbf) [inline playwright/jest-setup.js:9]: Do you think it makes sense to add logging to this so people can assert whether their configuration was working correctly? Maybe shown in certain log level (I guess something for the future)

**Review by yannbf — APPROVED** [review]
Looks great!! I added a few small comments

File: README.md (hunk 2 above)
Comment (shilman) [reply README.md:290]: Addding

File: README.md (hunk 3 above)
Comment (shilman) [reply README.md:23]: I use a VSCode addon that indents the ToC hierarchically. I think it reads better, and am tweaking it more in a followup commit. Open to other conventions though!

File: playwright/jest-setup.js (hunk 4 above)
Comment (yannbf) [reply playwright/jest-setup.js:9]: makes sense

File: README.md (hunk 3 above)
Comment (yannbf) [reply README.md]: sounds good!
